internal/service/ecs/cluster_capacity_providers_test.go

```diff
@@ -0,0 +1,412 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckClusterExists("aws_ecs_cluster.test", &cluster),
+					resource.TestCheckResourceAttr(resourceName, "default_capacity_provider_strategy.#", "1"),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "default_capacity_provider_strategy.*", map[string]string{
+						"base":              "1",
+						"weight":            "100",
+						"capacity_provider": "FARGATE",
+					}),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccClusterCapacityProvidersConfig_withDefaultCapacityProviderStrategy2(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckClusterExists("aws_ecs_cluster.test", &cluster),
+					resource.TestCheckResourceAttr(resourceName, "default_capacity_provider_strategy.#", "2"),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "default_capacity_provider_strategy.*", map[string]string{
+						"base":              "1",
+						"weight":            "50",
+						"capacity_provider": "FARGATE",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "default_capacity_provider_strategy.*", map[string]string{
+						"base":              "",
+						"weight":            "50",
+						"capacity_provider": "FARGATE_SPOT",
+					}),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccClusterCapacityProvidersConfig_withDefaultCapacityProviderStrategy3(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckClusterExists("aws_ecs_cluster.test", &cluster),
+					resource.TestCheckResourceAttr(resourceName, "default_capacity_provider_strategy.#", "2"),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "default_capacity_provider_strategy.*", map[string]string{
+						"base":              "2",
+						"weight":            "25",
+						"capacity_provider": "FARGATE",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "default_capacity_provider_strategy.*", map[string]string{
+						"base":              "",
+						"weight":            "75",
+						"capacity_provider": "FARGATE_SPOT",
+					}),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccClusterCapacityProvidersConfig_withDefaultCapacityProviderStrategy4(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckClusterExists("aws_ecs_cluster.test", &cluster),
+					resource.TestCheckResourceAttr(resourceName, "default_capacity_provider_strategy.#", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccClusterCapacityProvidersConfig(rName string) string {
```

Comment:
Would it be beneficial to add a test that mirrors [this functionality](https://github.com/hashicorp/terraform-provider-aws/issues/11409#issuecomment-1020666564) but using the `aws_ecs_cluster_capacity_providers`?